### PR TITLE
Improvements for API & CLI test automation

### PR DESCRIPTION
**Pull request by Daniel-VDM**
* Add option to specify test directory
* Add option to specify Cx src and dst endpoints
* Remove chain-id param
* Add option to specify max number of times to attempt test
* Fix localnet tests
* Add localnet Validator keys
* Add non-explorer test
* Add only-explorer test
* General bug fix
* Update documentation

### CODE REVIEW

File: test-automation/api-tests/test.py [hunk 1]
```diff
@@ -46,83 +59,181 @@ def load_keys(cli) -> list:
         for file in key_content:
             if file.endswith(".key"):
                 key_file_path = f"{os.getcwd()}/{args.keys_dir}/{key}/{file}"
-                new_account_name = f"api_test_{i}"
-                cli.remove_address(new_account_name)
+                new_account_name = f"_test_key_{random_num}_{i}"
+                CLI.remove_address(new_account_name)
```
Comment (fxfactorial) [inline test-automation/api-tests/test.py]: let's make the CLI remove keys, later make this consume that hmy keys --remove (something like that)

File: test-automation/api-tests/Jenkinsfile (nightly) [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+def build_webhookURL = 'https://discordapp.com/api/webhooks/568130604823085071/fJTFA9C7YNEkcsr-scZL9Q3xByqzfRxQj31klmP7nGU7Aa-4hIFlpg1l6vKg0oGzZOVM'
+
+node {
+  try {
+    stage('clean') {
+      if (params.CLEAN){
+        cleanWs()
+      }
+    }
+    docker.image('danielharmony/hmy-reg-test').inside('-u root:sudo -e "GOPATH=${WORKSPACE}" -e "XDG_CACHE_HOME="/tmp/.cache"') {
```
Comment (LeoHChen) [inline test-automation/api-tests/Jenkinsfile (nightly)]: why don't use harmony/ ? I think we own harmonyone namespace.

File: test-automation/api-tests/LocalnetValidatorKeys/s1/one1est2gxcvavmtnzc7mhd73gzadm3xxcv5zczdtw.key [hunk 3]
```diff
@@ -0,0 +1 @@
+{"address":"cc16a41b0ceb36b98b1edddbe8a05d6ee2636194","crypto":{"cipher":"aes-128-ctr","ciphertext":"6d7230bf642757bef746e720a9598803748ce99bc7052ee502dfd992eb160f59","cipherparams":{"iv":"081dec8badd657b1cf0d8c1c748379ad"},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"254cecddedc1e238d83c4064fac06ae92467d0ddf882b2c506c951f494967b36"},"mac":"73e67e5781943ee42e73892ca3c3e1e9624a2d1ec6881716fcf3f16f249d9d75"},"id":"c87bc580-938a-43d9-b700-af6360fbd6aa","version":3}
```
Comment (LeoHChen) [inline test-automation/api-tests/LocalnetValidatorKeys/s1/one1est2gxcvavmtnzc7mhd73gzadm3xxcv5zczdtw.key]: don't submit any key in the PR, even it is in testnet.

File: test-automation/api-tests/test.py [hunk 4]
```diff
@@ -46,83 +59,181 @@ def load_keys(cli) -> list:
...
         to_addr_candidates.remove(acc_name)
         to_addr = cli.get_address(random.choice(to_addr_candidates))
         if balances[source_shard]["amount"] >= 1e-9:
+            print(f"Raw transaction details:\n"
+                  f"\tNode: {node}\n"
+                  f"\tFrom: {from_addr}\n"
+                  f"\tTo: {to_addr}\n"
+                  f"\tFrom-shard: {source_shard}\n"
+                  f"\tTo-shard: {1-source_shard}")
             if chain_id not in {"mainnet", "testnet", "pangaea"}:  # Must be localnet
```
Comment (fxfactorial) [inline test-automation/api-tests/test.py]: make this set come from CLI dumping what chain-ids it knows about

File: test-automation/cli-tests/tests/testHmy.py [hunk 5]
```diff
@@ -42,77 +28,172 @@ def load_environment():
...
+        LOGGER(f"Failed: Balance for reference is {request_bal} but need at least {ref_min_bal} for test to be valid.")
+        return False
 
-def get_address_from_name(name):
-    if name in ADDRESSES:
-        return ADDRESSES[name]
-    else:
-        load_addresses()
-        return ADDRESSES.get(name, None)
+    cli_s0_bal = round(cli_response_list[0]["amount"], 6)
```
Comment (fxfactorial) [inline test-automation/cli-tests/tests/testHmy.py]: double check the determinism of this rounding logic, fix in upstream hmy CLI if need be

File: test-automation/cli-tests/tests/utils.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            logging.warning(final_msg)
+            print(f"[ERROR] {final_msg}")
+        else:
+            print(final_msg)
+
+    return log
+
+
+def test_announce(fn):
+    def warp(*args):
```
Comment (fxfactorial) [inline test-automation/cli-tests/tests/utils.py]: typo

**Pull request merged**